unittest/joint-generic.cpp

```diff
@@ -75,7 +75,122 @@ void test_joint_methods(
     jda.S().matrix().isApprox(jdata.S().matrix()),
     std::string(error_prefix + " - JointMotionSubspaceXd "));
   BOOST_CHECK_MESSAGE(
-    (jda.M()).isApprox((jdata.M())),
+    (jda.M()).isApprox((jdata.M()), 1e-6),
+    std::string(error_prefix + " - Joint transforms ")); // ==  or isApprox ?
+  BOOST_CHECK_MESSAGE(
+    (jda.v()).isApprox((pinocchio::Motion(jdata.v()))),
+    std::string(error_prefix + " - Joint motions "));
+  BOOST_CHECK_MESSAGE((jda.c()) == (jdata.c()), std::string(error_prefix + " - Joint bias "));
+
+  BOOST_CHECK_MESSAGE(
+    (jda.U()).isApprox(jdata.U()),
+    std::string(error_prefix + " - Joint U inertia matrix decomposition "));
+  BOOST_CHECK_MESSAGE(
+    (jda.Dinv()).isApprox(jdata.Dinv()),
+    std::string(error_prefix + " - Joint DInv inertia matrix decomposition "));
+  BOOST_CHECK_MESSAGE(
+    (jda.UDinv()).isApprox(jdata.UDinv()),
+    std::string(error_prefix + " - Joint UDInv inertia matrix decomposition "));
+
+  // Test vxS
+  typedef typename JointModel::Constraint_t Constraint_t;
+  typedef typename Constraint_t::DenseBase ConstraintDense;
+
+  Motion v(Motion::Random());
+  ConstraintDense vxS(v.cross(jdata.S()));
+  ConstraintDense vxS_ref = v.toActionMatrix() * jdata.S().matrix();
+
+  BOOST_CHECK_MESSAGE(vxS.isApprox(vxS_ref), std::string(error_prefix + "- Joint vxS operation "));
+
+  // Test Y*S
+  const Inertia Isparse(Inertia::Random());
+  const Inertia::Matrix6 Idense(Isparse.matrix());
+
+  const ConstraintDense IsparseS = Isparse * jdata.S();
+  const ConstraintDense IdenseS = Idense * jdata.S();
+
+  BOOST_CHECK_MESSAGE(
+    IdenseS.isApprox(IsparseS), std::string(error_prefix + "- Joint YS operation "));
+
+  // Test calc
+  {
+    JointData jdata1(jdata.derived());
+
+    jmodel.calc(jdata1.derived(), q1, v1);
+    jmodel.calc(jdata1.derived(), Blank(), v2);
+
+    JointData jdata_ref(jdata.derived());
+    jmodel.calc(jdata_ref.derived(), q1, v2);
+
+    BOOST_CHECK_MESSAGE(
+      pinocchio::JointData(jdata1).v() == pinocchio::JointData(jdata_ref).v(),
+      std::string(error_prefix + "- joint.calc(jdata,*,v) "));
+  }
+}
+
+template<typename Scalar, int Options>
+void test_joint_methods(
+  JointModelBase<JointModelSplineTpl<Scalar, Options>> & jmodel,
+  JointDataBase<typename JointModelSplineTpl<Scalar, Options>::JointDataDerived> & jdata)
+{
+  typedef typename LieGroup<JointModelSplineTpl<Scalar, Options>>::type LieGroupType;
+  typedef typename JointModelSplineTpl<Scalar, Options>::JointDataDerived JointData;
+
+  std::cout << "Testing Joint over " << jmodel.shortname() << std::endl;
+
+  Eigen::VectorXd q1, q2;
+  Eigen::VectorXd armature =
+    Eigen::VectorXd::Random(jdata.S().nv()) + Eigen::VectorXd::Ones(jdata.S().nv());
+
+  q1 = LieGroupType().randomConfiguration(
+    Eigen::VectorXd::Zero(jmodel.nq()), Eigen::VectorXd::Ones(jmodel.nq()));
+  q2 = LieGroupType().randomConfiguration(
+    Eigen::VectorXd::Zero(jmodel.nq()), Eigen::VectorXd::Ones(jmodel.nq()));
```

Comment:
Same here, randomConfiguration is the only difference between generic and spline joint.
If there is no way to specialize randomConfiguration, we should add a mechanism to this test to generate a valid randomConfiguration without duplicating the test.